### ISSUE TITLE
Visual Basic only reports BC37207 (conflicting attribute/option for signing) for netmodules

### ISSUE THREAD

**Issue by khyperia** [dotnet/roslyn#11427]
Using the test VB code from `InternalsVisibleToAndStrongNameTests.vb`, `SignModuleKeyContainerCmdLine_2()`:

test.vb:

```
<Assembly: System.Reflection.AssemblyKeyName("bogus")>

Public Class C
End Class
```

Run:

```
> vbc.exe /target:module /keyfile:someKey.snk test.vb
vbc : error BC37207: Attribute 'System.Reflection.AssemblyKeyFileAttribute' given in a source file conflicts with option 'CryptoKeyFile'.
> vbc.exe /target:library /keyfile:someKey.snk test.vb
// success, no diagnostics. Attribute ignored (not even checked, 'bogus' doesn't exist but 'someKey.snk' does)
// also reproducable if someKey.snk doesn't exist, but then we get
// `vbc : error BC36980: Error extracting public key from file 'someKey.snk': File not found.`
// (which is still not the attribute conflict error)
```

Expected behavior:

BC37207 is reported for all target types.

Issue was discovered in #11334, and blocked adding the new warning `WRN_AttributeIgnoredWhenPublicSigning` to VB. When fixing this issue, add that warning too (copy-translating the same check from C# to VB in the same corresponding method should work).

We could replicate C#'s behavior, but there's potential issues with backcompat, which is why I'm filing a new issue instead of fixing it in the above PR. There's a few courses of action:
1. Change the error to be shown for all types.
2. Don't change the error, but add the new warning to all types.
3. Don't change the error, and only add the new warning to netmodules.
